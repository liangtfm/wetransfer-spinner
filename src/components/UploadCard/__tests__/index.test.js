import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

import { UploadCard } from "..";

describe("UploadCard", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<UploadCard />);

    expect(asFragment(<UploadCard />)).toMatchSnapshot();
    expect(screen.getByText(/Add your files/i)).toBeInTheDocument();
    expect(screen.getByRole("button", /Transfer/i)).toBeInTheDocument();
    expect(screen.queryByText(/0%/i)).toBeNull();
  });

  it("shows spinner when transfer button is clicked", () => {
    render(<UploadCard />);
    const transferButton = screen.getByRole("button", /Transfer/i);

    fireEvent.click(transferButton);

    expect(screen.getByText(/0%/i)).toBeInTheDocument();
    expect(screen.getByText("Transferring...")).toBeInTheDocument();
    expect(screen.getByRole("button", /Cancel/i)).toBeInTheDocument();
  });

  it("shows done button when upload is complete", () => {
    render(<UploadCard progress={100} />);

    const transferButton = screen.getByRole("button", /Transfer/i);

    fireEvent.click(transferButton);

    expect(screen.getByText(/100%/i)).toBeInTheDocument();
    expect(screen.getByText(/Transfer Complete!/i)).toBeInTheDocument();
    expect(screen.getByRole("button", /Done/i)).toBeInTheDocument();
  });
});
