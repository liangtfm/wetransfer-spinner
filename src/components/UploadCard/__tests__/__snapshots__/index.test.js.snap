// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-jSgupP iSEwmU"
  >
    <div
      class="sc-hKgILt kYWFQG"
    >
      <div
        class="sc-jrAGrp hTWpAL"
      >
        Add your files
      </div>
    </div>
    <div
      class="sc-eCssSg fCoqQj"
    >
      <button
        class="sc-bdfBwQ sc-gsTCUz gmIHCr caSOrm"
        role="button"
        type="button"
      >
        Transfer
      </button>
    </div>
  </div>
</DocumentFragment>
`;
